lms/static/js/student_account/views/FinishAuthView.js

```diff
@@ -152,13 +155,34 @@
                 }
             },
 
+            /**
+             * Check if the target url is same as current domain or a subdomain.
+             * @param url Target url
+             * @returns {boolean}
+             */
+            isAllowedOrigin: function( url ) {
+                url = !url.includes('http') ? location.protocol + '//' + url : url;
+                var target = new URL(url);
+                var splittedDomain = location.host.split('.');
+                var domain = splittedDomain[splittedDomain.length-2] + '.' + splittedDomain[splittedDomain.length-1]
+                return target.host.includes(domain);
+            },
+
             /**
              * Redirect to a URL.  Mainly useful for mocking out in tests.
              * @param  {string} url The URL to redirect to.
              */
             redirect: function( url ) {
-                this.updateTaskDescription(gettext("Loading your courses"));
-                window.location.replace(url);
+                // Handle redirection to other origins (Edraak Programs/Marketing site) if 'next_origin' parameter exist.
+                if (this.nextOrigin && this.isAllowedOrigin(this.nextOrigin)) {
+                    this.nextOrigin = !this.nextOrigin.includes('http') ? location.protocol + '//' + this.nextOrigin : this.nextOrigin;
+                    var redirectURL = new URL(url, this.nextOrigin);
+                    window.location.replace(redirectURL.href);
+
+                } else {
+                    this.updateTaskDescription(gettext("Loading your courses"));
+                    window.location.replace(url);
```

Comment:
This won't work for urls that end with `.co.uk` for example. So, you have to make certain assumptions. For example what are the domains/sub-domains that will talk with this one, and accordingly create a function that solves our problem not the world's problem.